### DISCUSSION TITLE
ltsp dnsmasq  proper configuration with one NIC

### DISCUSSION THREAD

**Discussion by doguibnu** [ltsp/ltsp#647] [General]
Hello!
How are you?

I made LTSP with POP OS 20.04 LTSP (ubuntu based). Its nice and ok!
But, I always stay confused when config ltsp dnsmask. 
The scenario is: Internet by provider on dhcp mode, one NIC and one DNS
So, at the NIC IP: 192.168.17/24 - Gateway: 192.168.1.1 and DNS: 177.x.x.x
and add other IP for ltsp: 192.168.67.1/24 on the same NIC

What the right proper configuration ltsp dnsmasq command?
ltsp dnsmasq -d0  -p1 --dns-server="my-dns1 my-dns2"

or
ltsp dnsmasq -d1 -p0 --dns-server="0.0.0.0 8.8.8.8 208.67.222.222"

Thank you

Douglas

**Comment by alkisg** [comment]
Do not add a second IP on the same NIC, it will confuse you and not help you.
You mistyped your primary IP, I suppose it is 192.168.1.7/24.
Remove the 192.168.67.1/24 IP.
Then run `sudo ltsp dnsmasq`.
That's all. The proxyDHCP server doesn't send the DNS address, your router sends it, so there's no need to define it anywhere.

**Reply by TeamErr** [reply]
what about only 1 ethernet, and the network using static addresses?

**Reply by alkisg** [reply]
PXE network booting requires a DHCP server.
You can have DHCP reservations (like static IP addresses), but you do need a DHCP server.
Is your DHCP server the LTSP server or not?

**Reply by TeamErr** [reply]
the dhcp server is not enabled on the router sir, what if the dhcp server is on the ltsp server with the appropriate ip range on the router's network? I have tried changing the ltsp-dnsmasq.conf ip range instead of the default ltsp, for example 192.168.10.2-192.168.10.253 but the client is unable to get the ip address. *note that the ltsp server only has 1 network card

**Reply by alkisg** [reply]
Suppose your router is 192.168.10.1.
And that your LTSP server is 192.168.10.10.
And that you have static IPs from 192.168.10.1 to 192.168.10.99.
And that you have dynamic IPs from 192.168.10.100 to 192.168.10.250.

Then, create a file called /etc/dnsmasq.d/local.conf with the following content:

```
# IP ranges to hand out
dhcp-range=192.168.10.100,192.168.10.250,12h
dhcp-option=option:router,192.168.10.1
```

Finally, run that command:

```
sudo ltsp dnsmasq -p0
```

**Reply by TeamErr** [reply]
every time I run the command the configuration that has been saved returns to the origin so I just restart the dnsmasq service and the client still doesn't get an ip addres.

Feb 17 16:03:17 ltsp-server systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
Feb 17 16:03:17 ltsp-server dnsmasq[7056]: dnsmasq: syntax check OK.
Feb 17 16:03:17 ltsp-server dnsmasq[7065]: started, version 2.80 DNS disabled
Feb 17 16:03:17 ltsp-server dnsmasq[7065]: compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset auth nettlehash DNSSEC loop-detect inotify dumpfile
Feb 17 16:03:17 ltsp-server dnsmasq-dhcp[7065]: DHCP, proxy on subnet 192.168.10.1
Feb 17 16:03:17 ltsp-server dnsmasq-dhcp[7065]: DHCP, IP range 192.168.11.20 -- 192.168.11.50, lease time 12h
Feb 17 16:03:17 ltsp-server dnsmasq-tftp[7065]: TFTP root is /var/lib/tftpboot
Feb 17 16:03:17 ltsp-server systemd[1]: Started dnsmasq - A lightweight DHCP and caching DNS server.
Feb 17 16:03:18 ltsp-server dnsmasq-dhcp[7065]: PXE(enp3s0) b8:ae:ed:7d:4c:5e proxy
Feb 17 16:06:50 ltsp-server dnsmasq-dhcp[7065]: message repeated 17 times: [ PXE(enp3s0) b8:ae:ed:7d:4c:5e proxy]
Feb 17 16:07:22 ltsp-server dnsmasq-dhcp[7065]: PXE(enp3s0) b8:ae:ed:7d:4c:5e proxy

**Reply by TeamErr** [reply]
> Misalkan router Anda adalah 192.168.10.1. Dan server LTSP Anda adalah 192.168.10.10. Dan Anda memiliki IP statis dari 192.168.10.1 hingga 192.168.10.99. Dan Anda memiliki IP dinamis dari 192.168.10.100 hingga 192.168.10.250.
> 
> Kemudian, buat file bernama /etc/dnsmasq.d/local.conf dengan konten berikut:
> 
> ```
> # IP ranges to hand out
> dhcp-range=192.168.10.100,192.168.10.250,12h
> dhcp-option=option:router,192.168.10.1
> ```
> 
> Terakhir, jalankan perintah itu:
> 
> ```
> sudo ltsp dnsmasq -p0
> ```

sorry sir there is something left sorry this is very successful

**Reply by alkisg** [reply]
In your output, I see both 192.168.10.x and 192.168.11.x, so you've done some mistake. Please re-read my instructions above. If you still have issues, come to the [chat room](https://ltsp.org/guides/chat-room/) for live support.

**Reply by TeamErr** [reply]
It worked, sir, now how do you make sure the IP of the client computer doesn't change?

**Reply by alkisg** [reply]
You need to read the dnsmasq.conf man page.
That part is not documented in LTSP, it's documented in dnsmasq.
Create static entries in your /etc/dnsmasq.d/local.conf using dhcp-hostsfile or a similar directive.
